### ISSUE TITLE
Generated directive doesn't use enumValues

### ISSUE THREAD

**Issue by rebolyte** [dotansimha/graphql-code-generator#3264]
**Describe the bug**

With an external enum defined in `enumValues` in my config file, the generated directive types do not use the imported value. It imports the enum correctly and uses it elsewhere, but not for directives. In the generated types:

```ts
// this is correct...
import { UserRole } from '@org/package';

// ...and gets used here...
export type GqlUser = {
   __typename?: 'User',
  id: Scalars['ID'],
  username: Scalars['String'],
  email: Scalars['String'],
  role: UserRole, // checks out
};

// ...but not here
export type GqlAuthDirectiveResolver<Result, Parent, ContextType = any, Args = {   role?: Maybe<Maybe<GqlUserRole>> }> = DirectiveResolverFn<Result, Parent, ContextType, Args>;
// (references `GqlUserRole` rather than imported `UserRole`)
```

This causes a TS compilation error: `error TS2304: Cannot find name 'GqlUserRole'`.

**To Reproduce**

1. My GraphQL schema:

```graphql
directive @auth(
    role: UserRole = ADMIN
) on OBJECT | FIELD_DEFINITION | ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION

enum UserRole {
    ADMIN
    USER
}

schema {
    query: Query
}

type Query {
    me: User!
}

type User {
    id: ID!
    username: String!
    email: String!
    role: UserRole!
}
```

2. My GraphQL operations:

n/a

3. My `codegen.yml` config file:

```yml
generates:
  resolvers.ts:
    config:
      typesPrefix: Gql
      maybeValue: T | undefined
      enumValues:
      	UserRole: '@org/package#UserRole'
    plugins:
      - typescript
      - typescript-resolvers
```

**Expected behavior**

The generated directive type would respect `enumValues` config, using the imported type.

I've worked around this for the moment with @graphql-codegen/add plugin by aliasing the type with `add: type GqlUserRole = UserRole;` in my config, but that's a hack.

Maybe this hasn't cropped up due to the combination of `enumValues` and `typesPrefix`?

Seems related to #2976.

**Environment:**

- OS: Win10
- `@graphql-codegen/cli`: 1.11.2
- `@graphql-codegen/typescript`: 1.11.2
- `@graphql-codegen/typescript-resolvers`: 1.11.2
- NodeJS: 12.8.1

**Comment by dotansimha** [comment]
I think I fixed it (and other issues in directives resolvers signature, like incorrect optional fields) in: https://github.com/dotansimha/graphql-code-generator/pull/3382 

@rebolyte , can you please try the latest alpha? `1.11.3-alpha-2219473f.89`

**Comment by dotansimha** [comment]
Fixed in `v1.12.0`

**Comment by rebolyte** [comment]
Success! Thanks @dotansimha 👏
